test/Test/Pos/Ssc/GodTossing/Toss/BaseSpec.hs

```diff
@@ -0,0 +1,573 @@
+    description_notVerifiesBadSig =
+        "unsuccessfully verifies a commitment signature and a mismatching epoch"
+    description_verifiesOkOpening =
+        "successfully verifies that an opening corresponding to the given commitment\
+\ does indeed belong to it"
+    description_notVerifiesBadOpening =
+        "unsuccessfully verifies a mismatching commitment and opening"
+    description_emptyPayload =
+        "an empty payload always leads to successful verification"
+    description_checksGoodCommsPayload =
+        "successfully checks payload of commitments with a proper epoch index"
+    description_checksBadCommsPayload =
+        "unsuccessfully checks payload of commitments with the right exception for each\
+        \ failure case"
+    description_checksGoodOpensPayload =
+        "successfully checks payload of openings when the global state is correct"
+    description_checksBadOpeningsPayload =
+        "unsuccessfully checks payload of openings with the right exception for each\
+        \ failure case"
+    description_checksGoodSharesPayload =
+        "successfully checks payload of shares"
+    description_checksBadSharesPayload =
+        "unsuccessfully checks payload of shares with the right exception for each\
+        \ failure case"
+    description_checksGoodCertsPayload =
+        "successfully checks valid payload of VSS certificates"
+    description_checksBadCertsPayload =
+        "unsuccessfully checks payload of certificates with the right exception for each\
+        \ failure case"
+
+verifiesOkComm :: CommitmentOpening -> Bool
+verifiesOkComm CommitmentOpening{..} =
+    verifyCommitment coCommitment
+
+notVerifiesBadComm :: BadCommitment -> Bool
+notVerifiesBadComm (getBadComm -> badComm) =
+    not . verifyCommitment $ badComm
+
+verifiesOkCommSig :: SecretKey -> Commitment -> EpochIndex -> Bool
+verifiesOkCommSig sk comm epoch =
+    let commSig = (toPublic sk, comm, sign SignCommitment sk (epoch, comm))
+    in verifyCommitmentSignature epoch commSig
+
+notVerifiesBadCommSig :: BadSignedCommitment -> EpochIndex -> Bool
+notVerifiesBadCommSig (getBadSignedC -> badSignedComm) epoch =
+    not $ verifyCommitmentSignature epoch badSignedComm
+
+verifiesOkOpening :: CommitmentOpening -> Bool
+verifiesOkOpening CommitmentOpening{..} =
+    verifyOpening coCommitment coOpening
+
+notVerifiesBadOpening :: BadCommAndOpening -> Bool
+notVerifiesBadOpening (getBadCAndO -> badCommsAndOp) =
+    not . uncurry verifyOpening $ badCommsAndOp
+
+emptyPayload
+    :: Monoid container
+    => (container -> ExceptT e PureToss a)
+    -> MultiRichmenStake
+    -> GtGlobalState
+    -> Bool
+emptyPayload pureToss mrs gtgs =
+    isRight $ tossRunner mrs gtgs $ pureToss mempty
+
+newtype GoodCommsPayload = GoodComms
+    { getGoodComms :: (EpochIndex, GtGlobalState, CommitmentsMap, MultiRichmenStake)
+    } deriving (Show, Eq)
+
+instance Arbitrary GoodCommsPayload where
```

Comment:
@gromakovsky I don't think I mentioned this in the meeting, but I think `Arbitrary` instances used only in tests should stay in the modules where they're used. These ones are especially large, and I think it'd just be adding noise to the signal, to try and differentiate between actual `Arbitrary` instances and test ones.

If you disagree, I can move them there, no problem with that as it's also a viable solution - keeping these instances here also has its problems.